pkg/cache/sql/integration_test.go

```diff
@@ -0,0 +1,324 @@
+package sql
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/rancher/lasso/pkg/cache/sql/informer"
+	"github.com/rancher/lasso/pkg/cache/sql/informer/factory"
+	"github.com/rancher/lasso/pkg/cache/sql/partition"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+const testNamespace = "sql-test"
+
+var defaultPartition = partition.Partition{
+	All: true,
+}
+
+type IntegrationSuite struct {
+	suite.Suite
+	testEnv   *envtest.Environment
+	clientset kubernetes.Clientset
+	restCfg   rest.Config
+}
+
+func (i *IntegrationSuite) SetupSuite() {
+	i.testEnv = &envtest.Environment{}
+	restCfg, err := i.testEnv.Start()
```

Comment:
I think it'd be good to check for the presence of the `KUBEBUILDER_ASSETS` environment variable early and fail if it's missing. Otherwise, the error message is confusing ("etcd is missing from kubebuilder binaries").

Reply:
I disagree here. The test script does some checking here to make sure that things are properly setup. If you choose to use something else, I think that's on you (as the developer) to debug. If we start integrity checking this, it's tough for us to choose when to stop (i.e. what if someone just sets this to a blank string? What if they set filepaths to random places that don't include the binaries? What if the binaries are there but they aren't actually the real programs?).

Reply:
KUBEBUILDER_ASSETS is definitive https://github.com/kubernetes-sigs/controller-runtime/blob/38546806f2faf5973e3321a7bd5bb3afdbb5767d/pkg/internal/testing/process/bin_path_finder.go#L58-L61

But other options are possible, so it definitely seems reasonable to not check for it.

Reply:
It's for developer experience - if I just click and run this test suite, I won't understand the error message. I wouldn't know that I have to run this test in a special way - via this script, and that script also runs other tests, which I may not want.

Reply:
Right, which is on you as the developer to figure out. If you'd like I can add a comment about the requirement to the file so it's immediately apparent on failure. However, I can't agree on adding checks for this case.

Reply:
A README or a section in the README could help here. You could:
- Point the user to it when testEnv.Start() fails
- Point the user to it when `setup-envtest` is not installed (in the scripts/test.sh file, not in the test)

See https://github.com/rancher/rancher/tree/main/tests/controllers for an example in r/r.

Reply:
I've added information to the readme and pointed the user to it in the error message.